labcontrol/db/process.py

```diff
@@ -3783,3 +3790,407 @@ def _generate_amplicon_prep_information(self):
             data[curr_prep_sheet_id] = sio.getvalue()
 
         return data
+
+    def _get_metagenomics_data_for_prep(self):
+        """Gathers prep_info metadata for Metagenomics file generation
+
+        A support method for Metagenomics prep info file generation. This
+        method is only called by _generate_metagenomics_prep_information().
+        Gathers metadata used by above method and performs initial munging
+        for clarity.
+
+        Returns
+        -------
+        list: dict, each one representing a row of results.
+
+        #note: this fetchall() seemed appropriate, as we only expect to return
+        #several hundred results at most. This allows us to capture the results
+        #and clean them up before handing them off. This also allows us to
+        #refactor this query in time without touching the rest of the code.
+        """
+        inst_mdl, equipment, reagent = self._get_additional_prep_metadata()
+
+        sql = """
```

Comment:
Non-blocking, this makes me cringe, but I recognize that it may be a necessary evil right now. That, or I'm just not a strong database developer in which case, please ignore my novice cringing.